src/python/openvino_genai/py_openvino_genai.pyi

```diff
@@ -0,0 +1,1703 @@
+    
+        :param m_durations: Total durations for each generate call in microseconds.
+        :type m_durations: List[MicroSeconds]
+    
+        :param num_generated_tokens: Total number of tokens generated.
+        :type num_generated_tokens: int
+    
+        :param num_input_tokens: Total number of tokens in the input prompt.
+        :type num_input_tokens: int
+    """
+    def __init__(self) -> None:
+        ...
+    @property
+    def detokenization_durations(self) -> list[float]:
+        ...
+    @property
+    def generate_durations(self) -> list[float]:
+        ...
+    @property
+    def m_batch_sizes(self) -> list[int]:
+        ...
+    @property
+    def m_durations(self) -> list[float]:
+        ...
+    @property
+    def m_new_token_times(self) -> list[float]:
+        ...
+    @property
+    def m_times_to_first_token(self) -> list[float]:
+        ...
+    @property
+    def tokenization_durations(self) -> list[float]:
+        ...
+class Scheduler:
+    """
+    Scheduler for image generation pipelines.
+    """
+    class Type:
+        """
+        Members:
+        
+          AUTO
+        
+          LCM
+        
+          LMS_DISCRETE
+        
+          DDIM
+        
+          EULER_DISCRETE
+        
+          FLOW_MATCH_EULER_DISCRETE
+        """
+        AUTO: typing.ClassVar[Scheduler.Type]  # value = <Type.AUTO: 0>
+        DDIM: typing.ClassVar[Scheduler.Type]  # value = <Type.DDIM: 3>
+        EULER_DISCRETE: typing.ClassVar[Scheduler.Type]  # value = <Type.EULER_DISCRETE: 4>
+        FLOW_MATCH_EULER_DISCRETE: typing.ClassVar[Scheduler.Type]  # value = <Type.FLOW_MATCH_EULER_DISCRETE: 5>
+        LCM: typing.ClassVar[Scheduler.Type]  # value = <Type.LCM: 1>
+        LMS_DISCRETE: typing.ClassVar[Scheduler.Type]  # value = <Type.LMS_DISCRETE: 2>
+        __members__: typing.ClassVar[dict[str, Scheduler.Type]]  # value = {'AUTO': <Type.AUTO: 0>, 'LCM': <Type.LCM: 1>, 'LMS_DISCRETE': <Type.LMS_DISCRETE: 2>, 'DDIM': <Type.DDIM: 3>, 'EULER_DISCRETE': <Type.EULER_DISCRETE: 4>, 'FLOW_MATCH_EULER_DISCRETE': <Type.FLOW_MATCH_EULER_DISCRETE: 5>}
+        def __eq__(self, other: typing.Any) -> bool:
+            ...
+        def __getstate__(self) -> int:
+            ...
+        def __hash__(self) -> int:
+            ...
+        def __index__(self) -> int:
+            ...
+        def __init__(self, value: int) -> None:
+            ...
+        def __int__(self) -> int:
+            ...
+        def __ne__(self, other: typing.Any) -> bool:
+            ...
+        def __repr__(self) -> str:
+            ...
+        def __setstate__(self, state: int) -> None:
+            ...
+        def __str__(self) -> str:
+            ...
+        @property
+        def name(self) -> str:
+            ...
+        @property
+        def value(self) -> int:
+            ...
+    @staticmethod
+    def from_config(scheduler_config_path: os.PathLike, scheduler_type: Scheduler.Type) -> Scheduler:
```

Comment:
in C++ `scheduler_type: Scheduler.Type` is optional argument, should it be optional in Python as well? I.e. to have a default value

Reply:
Made optional, with default Scheduler.Type.AUTO as in c++.